DFReader.py

```diff
@@ -744,6 +847,9 @@ def _add_msg(self, m):
             mavtype = build_types.get(m.BU,None)
             if mavtype is not None:
                 self.mav_type = mavtype
+                # Reset the meta data, when mav type is updated
+                self.metadata.reset()
```

Comment:
should only reset when it actually changes

Reply:
I did some checks, and as speculated, this doesn't appear to be needed at all..... init_arrays, called by constructor will have set the mav_type for the file before anyone can try accessing any metadata. Therefore I propose to remove the lines.